app/molecule/molecule-android/src/main/kotlin/com/hedvig/android/molecule/android/MoleculeViewModel.kt

```diff
@@ -1,43 +1,52 @@
-package com.hedvig.android.core.ui
+package com.hedvig.android.molecule.android
 
-import androidx.compose.runtime.Composable
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import app.cash.molecule.AndroidUiDispatcher
 import app.cash.molecule.RecompositionMode
 import app.cash.molecule.moleculeFlow
+import com.hedvig.android.molecule.public.MoleculePresenter
+import com.hedvig.android.molecule.public.MoleculePresenterScope
 import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.WhileSubscribed
+import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.flow.stateIn
 import kotlin.time.Duration.Companion.seconds
 
-abstract class MoleculeViewModel<Event, UiState> : ViewModel() {
+abstract class MoleculeViewModel<Event, Model>(
+  initialState: Model,
+  presenter: MoleculePresenter<Event, Model>,
+  val sharingStarted: SharingStarted = SharingStarted.WhileSubscribed(5.seconds),
+) : ViewModel() {
   private val scope = CoroutineScope(viewModelScope.coroutineContext + AndroidUiDispatcher.Main)
 
   // Events have a capacity large enough to handle simultaneous UI events, but
   // small enough to surface issues if they get backed up for some reason.
   private val events = MutableSharedFlow<Event>(extraBufferCapacity = 20)
 
+  private var seed: Model = initialState
```

Comment:
Can we rename this to something like `cachedState`, `storedState`, `lastState`? I think that better reflects what we are using it for

Reply:
Let's go with `lastState` since that's basically exactly what it is. The last state that this has ever seen. On initialization, the last state *is* technically the initial state, and from then on it's the last state that was emitted. Will fix!